uploader/src/main/scala/com/gu/media/UploadActionsLambda.scala

```diff
@@ -2,31 +2,101 @@ package com.gu.media
 
 import com.amazonaws.services.lambda.runtime.events.KinesisEvent
 import com.amazonaws.services.lambda.runtime.{Context, RequestHandler}
-import com.gu.media.aws.{S3Access, UploadAccess}
+import com.gu.media.aws.{DynamoAccess, S3Access, UploadAccess}
 import com.gu.media.lambda.LambdaBase
 import com.gu.media.logging.Logging
 import com.gu.media.upload._
+import com.gu.media.youtube.YouTubeAccess
+import com.squareup.okhttp._
 import play.api.libs.json.{JsError, JsSuccess, Json}
 
+import scala.collection.JavaConverters._
 import scala.util.control.NonFatal
 
 class UploadActionsLambda extends RequestHandler[KinesisEvent, Unit]
   with LambdaBase
   with S3Access
   with UploadAccess
+  with DynamoAccess
+  with YouTubeAccess
+  with YouTubeVideoUpload
+  with HmacRequestSupport
   with Logging {
 
+  private val table = new UploadsTable(this)
+  private val domain = sys.env.get("DOMAIN")
+
+  private val http = new OkHttpClient()
+  private val appJson = MediaType.parse("application/json")
+
   override def handleRequest(input: KinesisEvent, context: Context): Unit = {
     readAction(input).foreach {
       case UploadPartToYouTube(uploadId, key) =>
-        // TODO: upload to YouTube
-        log.info(s"$key uploaded for $uploadId")
+        uploadToYouTube(uploadId, key)
 
       case DeleteParts(uploadId, partsToDelete) =>
         deleteParts(partsToDelete)
     }
   }
 
+  private def uploadToYouTube(uploadId: String, partKey: String): Unit = {
+    getUploadAndPart(uploadId, partKey) match {
+      case Some((upload, part)) =>
+        val total = upload.parts.last.end
+        val uploadUri = getYTUploadUrl(upload)
+
+        log.info(s"Uploading ${part.key} [${part.start} - ${part.end}]")
+
+        uploadPart(uploadUri, part.key, part.start, part.end, total, (_: Long) => {}).foreach { videoId =>
+          log.info(s"Successful upload ${upload.id}. YouTube ID: $videoId")
+
+          addAsset(upload.metadata.atomId, videoId)
+          table.delete(upload.id)
+        }
+
+      case _ =>
+        log.error(s"Unknown upload id $uploadId or part $partKey")
+    }
+  }
+
+  private def getYTUploadUrl(upload: Upload): String = upload.youTube.upload.getOrElse {
+    log.info(s"Starting YouTube upload for ${upload.id} [${upload.metadata.title} - ${upload.youTube.channel}]")
+
+    val url = startUpload(upload.metadata.title, upload.youTube.channel, upload.id, upload.parts.last.end)
+    val updated = upload.copy(youTube = upload.youTube.copy(upload = Some(url)))
+
+    table.put(updated)
+    url
+  }
+
+  private def addAsset(atomId: String, videoId: String): Unit = {
+    val actuallyPerformRequest = stage != "DEV" && domain.nonEmpty
```

Comment:
I'd like to make it work in DEV, but since it's performing an HMAC request from a lambda running in AWS, I couldn't think of a way of making it hit the end-point (since I'm running the server locally).